common/step_create_cdrom.go

```diff
@@ -0,0 +1,201 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/hashicorp/packer/helper/builder/localexec"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/packer/tmp"
+)
+
+// StepCreateCD will create a CD disk with the given files.
+type StepCreateCD struct {
+	// Files can be either files or directories. Any files provided here will
+	// be written to the root of the CD. Directories will be written to the
+	// root of the CD as well, but will retain their subdirectory structure.
+	Files []string
+	Label string
+
+	CDPath string
+
+	filesAdded map[string]bool
+}
+
+func (s *StepCreateCD) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	if len(s.Files) == 0 {
+		log.Println("No CD files specified. CD disk will not be made.")
+		return multistep.ActionContinue
+	}
+
+	ui := state.Get("ui").(packer.Ui)
+	ui.Say("Creating CD disk...")
+
+	if s.Label == "" {
+		s.Label = "packer"
+	} else {
+		log.Printf("CD label is set to %s", s.Label)
+	}
+
+	// Track what files are added. Used for testing step.
+	s.filesAdded = make(map[string]bool)
+
+	// Create a temporary file to be our CD drive
+	CDF, err := tmp.File("packer*.iso")
+	// Set the path so we can remove it later
+	CDPath := CDF.Name()
+	CDF.Close()
+	os.Remove(CDPath)
```

Comment:
To confirm we generate a temp file here just to get the filename. It is then immediately removed because the ISO creation command will create an actual iso in this location. Is that correct?

Reply:
yep. There may be an easier way to achieve the same idea.